src/diagram/components/quantity-node.tsx

```diff
@@ -121,21 +138,18 @@ const _QuantityNode: React.FC<IProps> = ({ data, isConnectable }) => {
             </div>
           </div>
         }
-        <div className="variable-info-row">
-          <input className="variable-info value" type="number" placeholder="value" autoComplete="off" onChange={onValueChange} data-testid="variable-value"
-            value={shownValue !== undefined ? shownValue.toString() : ""} onMouseDown={e => e.stopPropagation()} />
-          <input className="variable-info unit" type="text" placeholder="unit" autoComplete="off" value={shownUnit|| ""} data-testid="variable-unit"
-            onChange={onUnitChange} onMouseDown={e => e.stopPropagation()} />
-        </div>
+        {variable.numberOfInputs >= 1 ? renderValueUnitUnEditable() : renderValueUnitInput()}
         <div>
-          <select className="variable-info operation" value={data.node.variable.operation || ""} onChange={onOperationChange}>
-          { // in an enumeration the keys are the names and the values are string or numeric identifier
+          {!(variable.numberOfInputs >= 1) &&
+            <select className="variable-info operation" value={data.node.variable.operation || ""} onChange={onOperationChange}>
+            { // in an enumeration the keys are the names and the values are string or numeric identifier
+            }
+              <option key="none" value="">none</option>
+              {Object.entries(Operation).map(([name, symbol]) =>
+                <option key={name} value={symbol} data-testid={`variable-operation-${name}`}>{symbol}</option>
+              )}
+            </select>
```

Comment:
This makes more sense than before. However there isn't a need for an operation if there are no inputs. So I think you can just remove this whole select element at this point.